var/spack/repos/builtin/packages/automake/package.py

```diff
@@ -2,13 +2,13 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
-
-from spack import *
-
 import os
 import re
 
+import spack.package
+
 
+@spack.package.detectable
```

Comment:
What does this get us? Does it speed things up when running `spack external find` or `spack external list`? Will this have problems like the ordering of `@run_after` and `@on_packages_attributes` if we ever add another decorator?

Reply:
This will just:
1. Attach a default implementation of `determine_spec_details` to the class
2. Register the class name in a list of detectable packages

The decorator doesn't depend on ordering with respect to other decorators. See https://github.com/spack/spack/pull/16526#issuecomment-630137390 for a command based on 2.